File: data/golang/chat_gm.go
package proto

import (
	"packet"
)

type ChatGm struct {
	Content                  string
}

func (this *ChatGm) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteString(this.Content)

	return pack.Encode(uint16(1530))
}

func (this *ChatGm) EncodeMsg() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteString(this.Content)

	return pack.ReadBytes()
}

func ChatGmDecode(pack *packet.Packet) *ChatGm {
	chatGm := &ChatGm{}

	chatGm.Content = pack.ReadString()
	return chatGm
}

File: data/typescript.cc/ChatGm.ts
import Packet from "@mi/mod/Packet";


export default class ChatGm {
	private _content: string;

	constructor(packet?: Packet) {
		if (packet) {
			this._content = packet.ReadString();
		}
	}

	public Encode(): Packet {
		let packet: Packet = new Packet();
		packet.WriteString(this._content);
		packet.Encode(2030);
		return packet;
	}

	public GetBuffer(): ByteBuffer {
		return this.Encode().GetBuffer();
	}

	public get content(): string { return this._content; }
	public set content(value: string) { this._content = value; }
}
